kodilla-good-patterns/src/main/java/com/kodilla/food2doors/ExtraFoodShop.java

```diff
@@ -0,0 +1,42 @@
+package com.kodilla.food2doors;
+
+public class ExtraFoodShop implements Shop {
+
+    private static final String ORDER_CREATOR_INFO = "Creating order for: %s %s , adress: %s. Product: %s, producer %s. Price: %s $, Quantity: %s, total order price: %s $";
+
+    private OrderRequest orderRequest;
+    private ExtraFoodShopRepository extraFoodShopRepository;
+    private String producerName = "ExtraFoodShop";
+
+    public ExtraFoodShop(OrderRequest orderRequest, ExtraFoodShopRepository extraFoodShopRepository) {
+        this.orderRequest = orderRequest;
+        this.extraFoodShopRepository = extraFoodShopRepository;
+    }
+
+    public OrderRequest getOrderRequest() {
+        return orderRequest;
+    }
+
+    public double countOrderPrice() {
+        return orderRequest.getQuantity() * orderRequest.getProduct().getProductPrice();
+    }
+
+    public boolean process() {
+        if (extraFoodShopRepository.check(orderRequest.getProduct(), orderRequest.getQuantity())) {
+            extraFoodShopRepository.remove(orderRequest.getProduct(), orderRequest.getQuantity());
+            System.out.println(String.format(ORDER_CREATOR_INFO,
```

Comment:
taki duzy String bardzo brzydko wyglada, niepeawdaz? Fajnie by bylo, gdybys mial jeszcze jakas zewnetrzna klase InformationService, ktora mialaby metode inform(OrderRequest request) i by robila to, co nalezy. Zobacz, ze po pierwsze w trzech klasach powtarzasz te sama stala ORDER_CREATOR_INFO. po drugie, zwiazujesz sie mocno z implementacja. Jesli teraz zamiast drukowania info na konsoli, trzeba by bylo wyslac maila, albo zapisac info do pliku, to trzy razy (w trzech klasach) musialbys robic takie same poprawki.